### ISSUE TITLE
[Bug]: [Accessibility] [Dropdown]The ‘Close’ button is not accessible with keyboard under 'Dropdown' component when user resizes the page up to 200%.

### ISSUE THREAD

**Issue by Jeffhooo** [microsoft/fluentui#29176]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz
    Memory: 12.96 GB / 31.72 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (115.0.1901.203)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

### Bug Description

##Pre-Requisites:
Navigate using keyboard only.
Screen Resolution: 1280*720
Scaling: 100%
Browser settings>>Zoom: 200%
Refresh the page.

##Repro Steps:
Open the [URL](https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown)
Navigate to "Basic Dropdowns" demo components.
Navigate to the first dropdown input box using tab key and activate using enter key.
Navigate to 'Close' button using tab key or arrow keys.
Observe the issue.

##Actual Result: 
‘Close’ button is not accessible with keyboard which appears when user resizes the page up to 200%. It is accessible only with a mouse. 

##Observation: 
List dialog is getting closed using Esc key.
  
##Expected Result: 
The focus of keyboard should move on the ‘Close’ button and on selecting with enter key/Space key, dialog should be closed.

##User Experience: 
The 'Close’ button is not accessible via keyboard; hence the user is not able to perform action on the controls, hence it is blocking the user from using the functionality.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
So, this is actually on purpose! The close button is primarily there for touch users and especially touch screen reader users, since they have no other methods of closing the modal dialog that appears on smaller screens.

Keyboard users have the option of hitting escape, which is the standard way to close dialogs and other modals. Since this modal is specifically for a combobox and is treated more as a listbox that happens to be visually displayed as a modal rather than a full dialog popup, we have opted to keep tab focus confined to the listbox until the user hits escape.
